src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -560,16 +561,16 @@ private BoundExpression TransformCompoundAssignmentLHS(BoundExpression originalL
                             var loweredArray = VisitExpression(arrayAccess.Expression);
                             var loweredIndices = VisitList(arrayAccess.Indices);
 
-                            return SpillArrayElementAccess(loweredArray, loweredIndices, stores, temps);
+                            return SpillArrayElementAccess(loweredArray, loweredIndices, stores, temps, localKind);
```

Comment:
>localKind); [](start = 104, length = 11)

I'm assuming that all the cases in this switch are covered by a nullable value type scenario. If not, let's add test coverage.

Reply:
Added array accesses. We had some for indexers, but not for regular arrays. Everything else should be covered already.

---
In reply to: [293121299](https://github.com/dotnet/roslyn/pull/36329#discussion_r293121299) [](ancestors = 293121299)